[API] Add a RBAC solution (#1115)
neicnordic/sensitive-data-archive: sda/cmd/api/api.md

    curl -H "Authorization: Bearer $token" -H "Content-Type: application/json" -X POST -d '{"pubkey": "'"$( base64 -w0 /PATH/TO/c4gh.pub)"'", "description": "this is the key description"}' https://HOSTNAME/c4gh-keys/add
    ```

#### Configure RBAC

> RBAC is configure according to the JSON schema below.

Reviewer: ```suggestion
RBAC is configured according to the JSON schema below.
```

The path to the JSON file containing the RBAC policies needs to be passed through the `api.rbacFile` config definition.

> The `policy` section will configure access to the defined endpoints, unless specific rules are set, an endpoint will not be accessible

Reviewer: ```suggestion
The `policy` section will configure access to the defined endpoints. Unless specific rules are set, an endpoint will not be accessible
```

> The `policy` section will configure access to the defined endpoints. Unless specific rules are set, an endpoint will not be accessible

Reviewer 2: ```suggestion
The `policy` section will configure access to the defined endpoints. Unless specific rules are set, an endpoint will not be accessible.
```

- `action`: can be single string value i,e `GET` or a regex string with `|` as separator i.e. `(GET)|(POST)|(PUT)`. In the later case all actions in the list are allowed.

> - `paht`: have two different wildcard notations `*`, matches any value and `:` that matches a specifc named value

Reviewer: ```suggestion
- `path`: the endpoint. Should be a string value with two different wildcard notations: `*` matches any value and `:` that matches a specific named value
```

> - `role`: is the role that will be able to access the path, `"*"` will match any role or user.

Reviewer: ```suggestion
- `role`: the role that will be able to access the path, `"*"` will match any role or user.
```
